core/autoscaling.go

```diff
@@ -620,7 +619,26 @@ func (a *autoScalingGroup) getAllowedInstanceTypes(baseInstance *instance) []str
 		return []string{baseInstance.typeInfo.instanceType}
 	}
 	return strings.Split(allowed, ",")
+}
+
+func (a *autoScalingGroup) getDisallowedInstanceTypes(baseInstance *instance) []string {
+	var disallowed, disallowedInstanceTypesTag string
+
+	// Check option of allowed instance types
```

Comment:
Maybe I got it wrong, but it's surprising to read `Check option of allowed instance types` while the `disallowed-instance-types` is then being checked.